### PR TITLE
update selfcal

### PR DESCRIPTION
**Pull request by yoachim**
Update `run_selfcal_metric` for `note` to `scheduler_note` schema update.
General updates to selfcal so filter is set properly.

### CODE REVIEW

File: rubin_sim/maf/run_selfcal_metric.py [hunk 1]
```diff
@@ -61,7 +72,7 @@ def run_selfcal_metric():
 
     # Make plots of the residuals map
     map_bundle = MetricBundle(
-        IdentityMetric(metric_name="PhotoCal Uniformity"),
+        IdentityMetric(metric_name="PhotoCal Uniformity, %s" % args.filter),
```
Comment (rhiannonlynne) [inline rubin_sim/maf/run_selfcal_metric.py:75]: Just to make things print nicer in the end, can you drop the ","? 
I admit this is just a matter of personal taste but  "Photocal Uniformity r <>" (and presumably maybe other bands ..)  just reads a bit easier than "Photocal Uniformity, r <>" ... 
I guess I don't care a ton but none of the other summary metrics use ","

File: rubin_sim/maf/run_selfcal_metric.py [hunk 2]
```diff
@@ -43,13 +45,22 @@ def run_selfcal_metric():
...
     # Exclude DDF visits
-    sql = "note not like '%DD%'"
+    sql = "scheduler_note not like '%DD%'"
     # And run on only year 1 (?)
     sql += " and night < 366"
-    bundle = MetricBundle(selfcal_metric, slicer, sql, run_name=sim_name, info_label="year 1 no-DD")
+    sql += " and filter='%s'" % args.filter
+    # Exclude strange exposure times
+    sql += " and visitExposureTime > 20"
+    sql += " and visitExposureTime < 40"
```
Comment (rhiannonlynne) [inline rubin_sim/maf/run_selfcal_metric.py:59]: Do we actually need to drop these? It's true it makes things neater, but I would have thought including these exposures wouldn't make the self-calibration worse, and these visits ought to be calibrated as well .. 
I'd also wonder about including the DDF visits, tbh, but I think they were excluded just for calculation speed perhaps.

File: rubin_sim/maf/maf_contrib/selfcal_uniformity_metric.py [hunk 3]
```diff
@@ -55,11 +62,10 @@ def __init__(self, nside_residual=128, highglat_cut=30.0, outlier_nsig=4.0):
         self.outlier_nsig = outlier_nsig
 
         self.units = "mmag"
+        self.filter_name = filter_name
 
     def run(self, data_slice, slice_point=None):
-        filter_name = data_slice["filter"][0]
```
Comment (rhiannonlynne) [inline rubin_sim/maf/maf_contrib/selfcal_uniformity_metric.py:60]: Well, that's interesting and I think I hadn't noticed. I don't think Eli had specified the constraint on the data either ... 
So, ha. Definitely should recheck the more downtime / less downtime outputs. 
I also wonder how well the metric tracks against the actual photometric calculation that Eli was expecting to do given that FGCM works across all filters (I think).

**Review by rhiannonlynne — APPROVED** [review]

File: rubin_sim/maf/run_selfcal_metric.py (hunk 2 above)
Comment (yoachim) [reply rubin_sim/maf/run_selfcal_metric.py:59]: I was just being extra conservative, but they should be able to be used. That will let in the twilight observations, but I think you're right that they will be weighted properly.

**Pull request merged**